qt/tools/genhooks_gui.py

```diff
@@ -240,14 +245,45 @@ def custom_link_handler(url):
     ),
     # Multiple windows
     ###################
+    # reviewer and previewer
+    Hook(
+        name="audio_will_replay",
+        args=["web_content: aqt.webview.AnkiWebView", "card: Card", "state: str"],
+        doc="""Called when the audio is replayed by the user.""",
+    ),
     # reviewer, clayout and browser
     Hook(
         name="card_will_show",
         args=["text: str", "card: Card", "kind: str"],
         return_type="str",
         legacy_hook="prepareQA",
+        doc="""Deprecated. Use card_will_show_state instead.
+        Can modify card text before review/preview.""",
+    ),
+    Hook(
+        name="card_will_show_state",
+        args=[
+            "text: str",
+            "card: Card",
+            "kind: str",
+            "web_content: aqt.webview.AnkiWebView",
+            "skip_front: bool",
+            "has_autoplayed: bool",
+        ],
+        return_type="str",
+        replaces="card_will_show",
+        replaced_hook_args=["text: str", "card: Card", "kind: str"],
```

Comment:
This is the correct way to change the signature of a hook, but I'm not so sure about this one:

- providing the webview seems reasonable, and ideally we'd have done this from the start. But there are a number of add-ons that use this hook now, and adding a new one will mean those existing add-on authors now need to update their add-ons and add extra code if they want to support older Anki versions. An alternate option would be to provide a function like get_webview_for_card_will_show_state(kind: str) -> webview, so that the signature doesn't need to be changed.
- skip_front and has_autoplayed are specific to the previewing and clayout screens respectively, and feel a bit out of place here. Maybe your add-on could check kind and extract that info from the relevant screens instead?

Reply:
I remove this new hook and created the `reviewer_did_init` and used other existing equivalent hooks to get the required information: . Here is how my addons works now: https://github.com/evandrocoan/AnkiAddons/commit/889bd1da94252c8ab7b46159e09046c4803fb4af

As this was the most difficult change, I did it first. The other changes I am going to rebase later this week.